tests/test_cli.py

```diff
@@ -212,10 +212,66 @@ def test_auth_handler_cookie():
     auth = CookieAuthHandler(identity=str(uuid.uuid4()))
     token = str(uuid.uuid4())
     with mock.patch(
-        "requests.Session.request",
-        side_effect=lambda *_, **__: mocked_auth_response("access_token", token)
+            "requests.Session.request",
+            side_effect=lambda *_, **__: mocked_auth_response("access_token", token)
     ):
         resp = auth(req)  # type: ignore
     assert "Authorization" not in resp.headers
     assert "Cookie" in resp.headers and len(resp.headers["Cookie"])
     assert resp.headers["Cookie"] == token
+
+
+@pytest.mark.cli
+def test_href_inputs_not_uploaded_to_vault():
+    mock_result = OperationResult(False, code=500)
+
+    def mock_upload(_href, *_, **__):
+        return mock_result
+
+    inputs = {"source": {"href": "https://fake.domain.com/fakefile.zip"}}
+    with mock.patch("weaver.cli.WeaverClient.upload", side_effect=mock_upload):
+        result = WeaverClient()._update_files(inputs=inputs)
+    assert result is not mock_result
```

Comment:
Add a message indicating explicitly that `upload` should not be called, since non-local file detection should skip this reference.